src/chunk.c

```diff
@@ -3674,22 +3674,28 @@ chunk_set_compressed_id_in_tuple(TupleInfo *ti, void *data)
 bool
 ts_chunk_set_compressed_chunk(Chunk *chunk, int32 compressed_chunk_id)
 {
+	bool success = false;
 	ScanKeyData scankey[1];
 	ScanKeyInit(&scankey[0],
 				Anum_chunk_idx_id,
 				BTEqualStrategyNumber,
 				F_INT4EQ,
 				Int32GetDatum(chunk->fd.id));
-	return chunk_scan_internal(CHUNK_ID_INDEX,
-							   scankey,
-							   1,
-							   chunk_check_ignorearg_dropped_filter,
-							   chunk_set_compressed_id_in_tuple,
-							   &compressed_chunk_id,
-							   0,
-							   ForwardScanDirection,
-							   RowExclusiveLock,
-							   CurrentMemoryContext) > 0;
+	success = chunk_scan_internal(CHUNK_ID_INDEX,
+								  scankey,
+								  1,
+								  chunk_check_ignorearg_dropped_filter,
+								  chunk_set_compressed_id_in_tuple,
+								  &compressed_chunk_id,
+								  0,
+								  ForwardScanDirection,
+								  RowExclusiveLock,
+								  CurrentMemoryContext) > 0;
+	if (success)
```

Comment:
@JamesGuthrie why did we change this? AFAICS, the chunk_set_compressed_id_in_tuple function will set the flag appropriately in the scan

Reply:
It does set it in the table, but it doesn't set it in `chunk` (it also can't, because it doesn't receive `chunk` as a parameter). When `ts_chunk_set_partial` is called later, it asserts that the chunk is compressed. That assertion is based on the `status` field of `chunk->fd`, not on the value in the table.

Reply:
Ah, so we update the catalog table but we don't update the entry inside the chunk. Got it.